### PR TITLE
Test doc for ngmodule template

### PR DESCRIPTION
**Pull request by jbogarthyde**
## PR Checklist
Please check if your PR fulfills the following requirements:

- [ ] The commit message follows our guidelines: https://github.com/angular/angular/blob/master/CONTRIBUTING.md#commit
- [ ] Tests for the changes have been added (for bug fixes / features)
- [x] Docs have been added / updated (for bug fixes / features)

## PR Type
What kind of change does this PR introduce?

```
[ ] Bugfix
[ ] Feature
[ ] Code style update (formatting, local variables)
[ ] Refactoring (no functional changes, no api changes)
[ ] Build related changes
[ ] CI related changes
[x] Documentation content changes
[ ] angular.io application / infrastructure changes
[ ] Other... Please describe:
```

## What is the current behavior?

Issue Number: N/A

## Does this PR introduce a breaking change?
```
[ ] Yes
[x] No
```

## Other information
Use to test API templates for NgModules, and as example for API doc training.
Minor style edits, move comments to correct locations.

### CODE REVIEW

File: packages/common/http/src/client.ts [hunk 1]
```diff
@@ -58,14 +62,21 @@ export class HttpClient {
...
-   * Construct a request which interprets the body as an `ArrayBuffer` and returns it.
+   * Constructs a request that interprets the body as an array buffer and returns it
+   * in an observable.
+   * @param method The HTTP method (such GET, PUT, or POST).
+   * @param url The URL to which to send the request.
+   * @param options An object containing the request body and options, as
+   * returned by an `addBody()` call.
    *
-   * @return an `Observable` of the body as an `ArrayBuffer`.
+   * @returns An observable of the request body.
```
Comment (trotyl) [inline packages/common/http/src/client.ts]: returns **request** body -> returns **response** body? (Same for other overloads below)

File: packages/common/http/src/client.ts [hunk 2]
```diff
@@ -58,14 +62,21 @@ export class HttpClient {
...
   request<R>(req: HttpRequest<any>): Observable<HttpEvent<R>>;
 
   /**
-   * Construct a request which interprets the body as an `ArrayBuffer` and returns it.
+   * Constructs a request that interprets the body as an array buffer and returns it
+   * in an observable.
+   * @param method The HTTP method (such GET, PUT, or POST).
+   * @param url The URL to which to send the request.
+   * @param options An object containing the request body and options, as
+   * returned by an `addBody()` call.
```
Comment (trotyl) [inline packages/common/http/src/client.ts]: `addBody()` is not part of public API yet, should it be mentioned in doc? (Same for other overloads below)

File: packages/common/http/src/client.ts [hunk 3]
```diff
@@ -18,8 +18,10 @@ import {HttpEvent, HttpResponse} from './response';
...
 
 /**
- * Construct an instance of `HttpRequestOptions<T>` from a source `HttpMethodOptions` and
- * the given `body`. Basically, this clones the object and adds the body.
+ * Appends a request body to a set of HTTP request options.
+ * @param options An `HttpMethodOptions` object that provides the request options.
+ * @param body The request body to add.
+ * @returns A new `HttpMethodOptions` object that includes the supplied request body.
  */
 function addBody<T>(
```
Comment (trotyl) [inline packages/common/http/src/client.ts]: Should it be exported to public API?

File: packages/common/http/src/module.ts [hunk 4]
```diff
@@ -137,11 +138,10 @@ export class HttpClientModule {
 }
 
 /**
- * `NgModule` which enables JSONP support in `HttpClient`.
- *
- * Without this module, Jsonp requests will reach the backend
- * with method JSONP, where they'll be rejected.
+ * An NgModule that enables JSONP support in `HttpClient`.
  *
+ * Provides interceptors that prevent Jsonp requests
```
Comment (trotyl) [inline packages/common/http/src/module.ts]: Consistency in case? JSONP/Jsonp.

File: packages/common/http/src/client.ts (hunk 3 above)
Comment (jbogarthyde) [reply packages/common/http/src/client.ts]: Question for the developer, not for doc. If it is exported, the doc is there.

File: packages/common/http/src/module.ts (hunk 4 above)
Comment (jbogarthyde) [reply packages/common/http/src/module.ts]: Haven't gotten to this yet - just did HttpClient NgModule, and overloads for get(), as a doc-generation test.

**Pull request closed without merging**